src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs

```diff
@@ -95,6 +94,30 @@ public virtual async Task<OrderHistoryViewModel> GetOrderHistoryViewModelAsync(G
             {
                 shipmentsTrackingInfos = await GetShipmentsTrackingInfoViewModels(orderQueryResult, param).ConfigureAwait(false);
             }
+            var orderSettings = await GetOrderSettings(param.Scope).ConfigureAwait(false);
+            
+            var fulfillmentOrdersList = new List<FulfillmentOrderQueryResult>();
+            if (param.OrderTense == OrderTense.CurrentOrders)
+            {
+                var fulfillmentLocationsToOrder = orderQueryResult.Results
+                    .Select(item => new
+                    {
+                        fulfillmentLocationIds = item.ShipmentItems.Select(el => el.FulfillmentLocationId).ToList(),
+                        orderId = item.Id
+                    }).ToList();
+
+                var fulfillmentLocationToOrder = fulfillmentLocationsToOrder
+                    .SelectMany(el => el.fulfillmentLocationIds.Select(item => new {orderId = el.orderId, fulfillmentLocationId = item}))
+                    .Distinct().ToList();
```

Comment:
this Distinct will not work, as this are objects and the references will be compared. You can remove it, and use Distinct later, when you have List of Order Guids (line 116)